nntrainer/layers/activation_layer.cpp

```diff
@@ -50,24 +50,19 @@ void ActivationLayer::forwarding(sharedConstTensors in) {
   Tensor &hidden_ = net_hidden[0]->getVariableRef();
   /// @note @a _act_fn is expected to work out of place and not modify @a input
   _act_fn(net_input[0]->getVariableRef(), hidden_);
```

Comment:
If activation function is not inplacable, this might run into a problem.

I think we should have:
1. type traits for inplacable operations
2. Deal each activation layer saperately.

For now, it looks goot to go for me.

Reply:
As of now, softmax activation layer does not work in-place.
For these kind of layers, the manager which is performing the in-place optimization is supposed to (and is currently also) ensure that those activation layers do not operate in-place as that will produce wrong result.

Reply:
check https://github.com/nnstreamer/nntrainer/pull/827/files#diff-8136da3afc181f3163a8f8a15d86d132abee8c07de2c9b49bc2cf0759e380a69R544
in-place optimizer function ensures that in-place optimization is only performed on layers which support in-place operations.

Later I will add a function for `layer` that would return if the `layer` can operate in-place. If yes, the in-place optimization function will make it in-place, else skip it.

Reply:
Sounds good